src/main/java/org/folio/inventory/resources/TenantItems.java

```diff
@@ -0,0 +1,137 @@
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.handler.BodyHandler;
+
+public class TenantItems {
+
+  private static final Logger log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+
+  private static final String TENANT_ITEMS_PATH = "/inventory/tenant-items";
+  public static final String ITEMS_FIELD = "items";
+  public static final String TOTAL_RECORDS_FIELD = "totalRecords";
+  public static final String TENANT_ID_FIELD = "tenantId";
+
+  private final HttpClient client;
+
+  public TenantItems(HttpClient client) {
+    this.client = client;
+  }
+
+  public void register(Router router) {
+    router.post(TENANT_ITEMS_PATH + "*").handler(BodyHandler.create());
+    router.post(TENANT_ITEMS_PATH).handler(this::getItemsFromTenants);
+  }
+
+  /**
+   *  This API is meant to be used by UI to fetch different items from several
+   *  tenants together within one call
+   *
+   */
+  private void getItemsFromTenants(RoutingContext routingContext) {
+    var getItemsFutures = routingContext.body().asPojo(TenantItemPairCollection.class)
+      .getTenantItemPairs().stream()
+      .collect(groupingBy(TenantItemPair::getTenantId, mapping(TenantItemPair::getItemId, toList())))
+      .entrySet().stream()
+      .map(tenantToItems -> getItemsWithTenantId(tenantToItems.getKey(), tenantToItems.getValue(), routingContext))
+      .toList();
+
+    CompletableFuture.allOf(getItemsFutures.toArray(new CompletableFuture[0]))
+      .thenApply(v -> getItemsFutures.stream()
+        .map(CompletableFuture::join)
+        .flatMap(List::stream)
+        .toList())
+      .thenApply(this::constructResponse)
+      .thenAccept(jsonObject -> JsonResponse.success(routingContext.response(), jsonObject));
+  }
+
+  private CompletableFuture<List<JsonObject>> getItemsWithTenantId(String tenantId, List<String> itemIds, RoutingContext routingContext) {
+    log.info("getItemsWithTenantId:: Fetching items - {} from tenant - {}", itemIds, tenantId);
+    var context = new WebContext(routingContext);
+    CollectionResourceClient itemsStorageClient;
+    try {
+      OkapiHttpClient okapiClient = createHttpClient(tenantId, context, routingContext);
+      itemsStorageClient = createItemsStorageClient(okapiClient, context);
+    }
+    catch (MalformedURLException e) {
+      invalidOkapiUrlResponse(routingContext, context);
+      return CompletableFuture.completedFuture(List.of());
+    }
+
+    var getByIdsQuery = multipleRecordsCqlQuery(itemIds);
+    var itemsFetched = new CompletableFuture<Response>();
+    itemsStorageClient.getAll(getByIdsQuery, itemsFetched::complete);
+
+    return itemsFetched.thenApplyAsync(response ->
+      getItemsWithTenantId(tenantId, response));
+  }
+
+  private List<JsonObject> getItemsWithTenantId(String tenantId, Response response) {
+    if (response.getStatusCode() != 200 || !response.hasBody()) {
```

Comment:
Please use an already-defined constant instead of a “magic number”.

Reply:
used HttpStatus enum